sources/Interop/Windows/Windows/other/helper-types/Windows.Manual.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace TerraFX.Interop.Windows;
+
+public static partial class Windows
+{
+    [DoesNotReturn]
+    public static void ThrowExternalException(string methodName, int errorCode)
+    {
+        var message = string.Format("'{0}' failed with an error code of '{1}'", methodName, errorCode);
+        throw new ExternalException(message, errorCode);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static void ThrowIfFailed(this HRESULT value, [CallerArgumentExpression("value")] string? valueExpression = null)
```

Comment:
For my uses, this is problematic -- I have my own `ThrowOnError()` extension method that goes through an `ExceptionFactory`. The HRESULT is converted into a specific exception, whether that's `IOException`, `ArgumentException`, or library-specific exceptions like `PaintDotNet.Direct2D1.RecreateTargetException` or `Animation.IllegalReentrancyException`. 

It's problematic because my muscle memory is now such that I type `hr.T` and press tab, which currently auto-completes to `hr.ThrowOnError()`. However, now it might auto-complete to `ThrowIfFailed()`, and it will be difficult to catch these mistakes. I do have `catch`'s where I go for the specific exception, not just `ExternalException`. So this is very important for my code base, whether writing new code or maintaining old code.

If there was a way to install an exception factory into TerraFX, then I could plug mine in and remove my own exception factory. There are probably other ways to solve this, too.

An example:
```cs
public ITransformedGeometry CreateTransformedGeometry(IGeometry source, Matrix3x2Float transform)
{
    using ComPtr<ID2D1Geometry> spSource = default;
    HRESULT hr = ComObject.GetIUnknown(source, &spSource);
    hr.ThrowOnError();

    using ComPtr<ID2D1TransformedGeometry> spTransformedGeometry = default;
    hr = this.pD2D1Factory->CreateTransformedGeometry(
        spSource,
        (D2D_MATRIX_3X2_F*)&transform,
        spTransformedGeometry.GetAddressOf());
    hr.ThrowOnError();

    return ComObjectRef.CreateRef<ITransformedGeometry>(spTransformedGeometry)!;
}
```

Reply:
I've dropped it being an extension method in favor of just being a static method here: https://github.com/terrafx/terrafx.interop.windows/pull/297

The alternative would have been for your `ThrowOnError` to be named the same (`ThrowIfFailed`) in which case it would be preferenced due to being a "closer" extension method